sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -870,9 +872,11 @@ def __init__(self, port_mapping):
         self.task_process = None
         self.port_dict = {}
         self.port_mapping = copy.deepcopy(port_mapping)
+        self.xcvr_table_helper = XcvrTableHelper()
```

Comment:
If `XcvrTableHelper` is implemented as a class member, it should be initialized in `task_worker` instead of `__init__`.
This is because `CmisManagerTask` is a subprocess wrapper, which means a subprocess will be forked once `task_run` is called.
If it is initialized in `__init__`, the flow is
- parent creates the `XcvrTableHelper` along with all the low level objects, like the sockets based on which the tables are accessed
- and then the child is forked, all resources opened by the parent are duplicated
- now there are two copies of all the resources but the parent's one will not really be used.

Reply:
Of course, initializing it in `__init__` and then duplicating it from parent to child should also work. But it looks like taking a detour to me.

Reply:
@stephenxs what is the concern here? only the worker thread is going to use the socket, not the parent.

Reply:
`xcvr_table_helper` is initialized in the parent and then duplicated in the client. but it will never be used by the parent.
so the resource allocated in the parent is wasted.
- if it is initialized in `__init__`: 
  - `xcvr_table_helper` initialized in parent. resource allocated for it but not used.
  - duplicated in the child during fork
- if it is initialized in `task_worker`
  - `xcvr_table_helper` is initialized in the child before it starts to work. 
  - no resource was allocated in the parent.

as I mentioned, it also works in the current way. so I'm ok if you do not decide to change it.